src/containers/ChangePasswordPage.jsx

```diff
@@ -0,0 +1,217 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this license. If not, see
+ *
+ *     https://www.gnu.org/licenses/
+ *
+ *
+ */
+
+import React from 'react';
+import intl from 'react-intl-universal';
+import ReactRouterPropTypes from 'react-router-prop-types';
+
+import './ChangePasswordPage.less';
+import { login } from '../utils/AuthService';
+import { cortexFetch } from '../utils/Cortex';
+
+const Config = require('Config');
+
+// Array of zoom parameters to pass to Cortex
+
+const zoomArray = [
+  'defaultprofile',
+  'defaultprofile:emails',
+  'defaultprofile:emails:element',
+  'defaultprofile:emails:element:list',
+  'defaultprofile:emails:element:profile',
+  'defaultprofile:emails:emailform',
+  'defaultprofile:emails:profile',
+  'defaultprofile:addresses:element',
+  'defaultprofile:addresses:billingaddresses:default',
+  'defaultprofile:paymentmethods',
+  'defaultprofile:paymentmethods:paymenttokenform',
+  'defaultprofile:paymentmethods:element',
+];
+
+class ChangePasswordForm extends React.Component {
+  static propTypes = {
+    location: ReactRouterPropTypes.location.isRequired,
+    history: ReactRouterPropTypes.history.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      oldPassword: '',
+      newPassword: '',
+      newPasswordConfirmed: '',
+      failedSubmit: false,
+      email: '',
+    };
+    this.setOldPassword = this.setOldPassword.bind(this);
+    this.setNewPassword = this.setNewPassword.bind(this);
+    this.setNewPasswordConfirmed = this.setNewPasswordConfirmed.bind(this);
+    this.submitNewPassword = this.submitNewPassword.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchProfileData();
+  }
+
+  setOldPassword(event) {
+    this.setState({ oldPassword: event.target.value });
+  }
+
+  setNewPassword(event) {
+    this.setState({ newPassword: event.target.value });
+  }
+
+  setNewPasswordConfirmed(event) {
+    this.setState({ newPasswordConfirmed: event.target.value });
+  }
+
+  cancel() {
+    const { location, history } = this.props;
+    if (location.state && location.state.returnPage) {
+      history.push(location.state.returnPage);
+    } else if (localStorage.getItem(`${Config.cortexApi.scope}_oAuthRole`) === 'REGISTERED') {
+      history.push('/profile');
+    } else {
+      history.push('/');
+    }
+  }
+
+  submitNewPassword(event) {
+    event.preventDefault();
+    const {
+      oldPassword, newPassword, newPasswordConfirmed, email,
+    } = this.state;
+    if (!oldPassword || !newPassword || !newPasswordConfirmed) {
+      this.setState({ failedSubmit: true });
+    }
+    login()
+      .then((response) => {
+        console.log(response);
+        cortexFetch(`/passwords/${Config.cortexApi.scope}/reset/form/?zoom=resetpasswordaction`, {
```

Comment:
Can we get this URL from the API rather than constructing it ourselves?

Reply:
Resolution of this is here https://github.com/elasticpath/react-pwa-reference-storefront/pull/267/commits/294b11874056bd3569c6ce239b5c27de1252f6d0